### PR TITLE
MODPERMS-229: Update deps for Q: RMB 35.2.0, Vertx 4.5.5, log4j 2.23.1

### PR DESCRIPTION
**Pull request by julianladisch**
https://folio-org.atlassian.net/browse/MODPERMS-229

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -196,7 +189,7 @@
       <plugin>
         <groupId>dev.aspectj</groupId>
         <artifactId>aspectj-maven-plugin</artifactId>
-        <version>1.13.1</version>
+        <version>1.14</version>
```
Comment (steveellis) [inline pom.xml:192]: Can we go to 1.15

File: pom.xml [hunk 2]
```diff
@@ -221,12 +214,12 @@
...
             <groupId>org.aspectj</groupId>
             <artifactId>aspectjrt</artifactId>
-            <version>1.9.20.1</version>
+            <version>1.9.21.1</version>
           </dependency>
           <dependency>
             <groupId>org.aspectj</groupId>
             <artifactId>aspectjtools</artifactId>
-            <version>1.9.20.1</version>
+            <version>1.9.21.1</version>
```
Comment (steveellis) [inline pom.xml:222]: Looks like there is a 1.9.21.2

File: pom.xml (hunk 1 above)
Comment (julianladisch) [reply pom.xml:192]: No:
```
[ERROR] Plugin dev.aspectj:aspectj-maven-plugin:1.15 or one of its dependencies could not be resolved: Could not find artifact dev.aspectj:aspectj-maven-plugin:jar:1.15 in folio-nexus (https://repository.folio.org/repository/maven-folio) -> [Help 1]
```
https://github.com/dev-aspectj/aspectj-maven-plugin/releases
https://mvnrepository.com/artifact/dev.aspectj/aspectj-maven-plugin

File: pom.xml (hunk 2 above)
Comment (julianladisch) [reply pom.xml:222]: That causes this warning:
```
[INFO] --- aspectj-maven-plugin:1.14:compile (default) @ mod-permissions ---
[INFO] Showing AJC message detail for messages of types: [error, warning, fail]
[WARNING] bad version number found in /home/aa/.m2/repository/org/aspectj/aspectjrt/1.9.21.1/aspectjrt-1.9.21.1.jar expected 1.9.21.2 found 1.9.21.1
```

File: pom.xml (hunk 1 above)
Comment (steveellis) [reply pom.xml:192]: Ok sorry I was looking at the one from org.codehaus.mojo which has a 1.15.

File: pom.xml (hunk 2 above)
Comment (steveellis) [reply pom.xml:222]: Noticed it here: https://mvnrepository.com/artifact/org.aspectj/aspectjtools.

**Review by steveellis — APPROVED** [review]

**Pull request merged**